taosWS/statement_test.go

```diff
@@ -0,0 +1,2159 @@
+package taosWS
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStmtExec(t *testing.T) {
+	db, err := sql.Open(driverName, dataSourceName)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer func() {
+		t.Log("start3")
+		db.Close()
+		t.Log("done3")
+	}()
+	defer func() {
+		_, err = db.Exec("drop database if exists test_stmt_driver_ws")
+		if err != nil {
+			t.Error(err)
+			return
+		}
+		t.Log("done2")
+	}()
+	_, err = db.Exec("create database if not exists test_stmt_driver_ws")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	_, err = db.Exec("create table if not exists test_stmt_driver_ws.ct(ts timestamp," +
+		"c1 bool," +
+		"c2 tinyint," +
+		"c3 smallint," +
+		"c4 int," +
+		"c5 bigint," +
+		"c6 tinyint unsigned," +
+		"c7 smallint unsigned," +
+		"c8 int unsigned," +
+		"c9 bigint unsigned," +
+		"c10 float," +
+		"c11 double," +
+		"c12 binary(20)," +
+		"c13 nchar(20)" +
+		")")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	stmt, err := db.Prepare("insert into test_stmt_driver_ws.ct values (?,?,?,?,?,?,?,?,?,?,?,?,?,?)")
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	result, err := stmt.Exec(time.Now(), 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, "binary", "nchar")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	affected, err := result.RowsAffected()
+	assert.NoError(t, err)
+	assert.Equal(t, int64(1), affected)
+	t.Log("done")
+}
+
+func TestStmtQuery(t *testing.T) {
+	db, err := sql.Open(driverName, dataSourceName)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer db.Close()
+	defer func() {
+		db.Exec("drop database if exists test_stmt_driver_ws_q")
+	}()
+	_, err = db.Exec("create database if not exists test_stmt_driver_ws_q")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	_, err = db.Exec("create table if not exists test_stmt_driver_ws_q.ct(ts timestamp," +
+		"c1 bool," +
+		"c2 tinyint," +
+		"c3 smallint," +
+		"c4 int," +
+		"c5 bigint," +
+		"c6 tinyint unsigned," +
+		"c7 smallint unsigned," +
+		"c8 int unsigned," +
+		"c9 bigint unsigned," +
+		"c10 float," +
+		"c11 double," +
+		"c12 binary(20)," +
+		"c13 nchar(20)" +
+		")")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	stmt, err := db.Prepare("insert into test_stmt_driver_ws_q.ct values (?,?,?,?,?,?,?,?,?,?,?,?,?,?)")
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	now := time.Now()
+	result, err := stmt.Exec(now, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, "binary", "nchar")
```

Comment:
can we insert many rows by call exec once?

Reply:
Only a single SQL statement is allowed